### DISCUSSION TITLE
suspended helmrelease triggers kustomize alert

### DISCUSSION THREAD

**Discussion by abohatyrenko** [fluxcd/flux2#4738] [Q&A] [answered]
Hello, when I'm suspending HelmRelease resource via spec.suspend: true, and it's getting applied I'm getting alert from kustomize-controller, which is a false positive

`Health check failed after 2m30.022317139s: timeout waiting for: [HelmRelease/namespace/app status: 'InProgress']`

I thought that kustomize-controller is supposed to read the changes on HelmRelease, is this expected and I need to somehow additionally patch the kustomize, including the fact that kustomize is syncing whole path:

```
  - spec:
      path: ./namespace/directory-with-helmreleases
```

**Comment by souleb** [comment]
We use [kstatus](https://github.com/kubernetes-sigs/cli-utils/blob/master/pkg/kstatus/README.md), to retrieve the status of the custom resources. When you suspend a `HelmRelease`, it can still have a status from previous or current reconciliation loop. When `kustomize-controller` performs a health check, it uses `kstatus` to poll deployed resources status. If those resources are `InProgress`, that will be returned. The fact that the resource may have by then been `suspended` is of no concern.

**Reply by stefanprodan** [reply, marked as answer]
When you suspend a Flux resources, kstatus will always fail since a change to `.spec.suspend` bumps the `generation` but the `observedGeneration` in status remains behind. See https://github.com/fluxcd/pkg/pull/765

**Reply by abohatyrenko** [reply]
Hello @stefanprodan, I see that the merge request is stuck at the draft stage, is there any ETA to expect a fix?

**Reply by stefanprodan** [reply]
No ETA, the fix is no good and my focus shifted to other Flux features.
